plugins/posthog/resoto_plugin_posthog/resources.py

```diff
@@ -106,8 +106,11 @@ def new(data: Dict) -> BaseResource:
         )
 
 
-def convert_date(date_str: str) -> Optional[datetime]:
+def convert_date(date_str: str = '') -> Optional[datetime]:
     try:
-        return datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+        if not isinstance(date_str, str):
```

Comment:
`date_str` is typed as str and then we check if the type is string.
This looks wrong. Do you want to prevent `None` or what other types can be expected?